test/framework/maven.go

```diff
@@ -0,0 +1,162 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/kiegroup/kogito-cloud-operator/test/config"
+)
+
+const (
+	mavenCommandName = "mvn"
+)
+
+// CreateMavenCommand methods initializes the basic data to run maven commands.
+func CreateMavenCommand(directory string) MavenCommand {
+	return &mavenCommandStruct{directory: directory}
+}
+
+// MavenCommand wraps information about the maven command to execute.
+type MavenCommand interface {
+	// WithLoggerContext method attaches a logger context to trace all the command logs when executing it.
+	WithLoggerContext(loggerContext string) MavenCommand
+	// Execute command and returns the outputs.
+	Execute(targets ...string) (string, error)
+
+	// SkipTests will skip testing automatically
+	SkipTests() MavenCommand
+	// Profiles sets the profiles to execute
+	Profiles(profiles ...string) MavenCommand
+}
+
+type mavenCommandStruct struct {
+	directory     string
+	loggerContext string
+
+	profiles     []string
+	otherOptions []string
+	skipTests    bool
+}
+
+func (mvnCmd *mavenCommandStruct) WithLoggerContext(loggerContext string) MavenCommand {
+	mvnCmd.loggerContext = loggerContext
+	return mvnCmd
+}
+
+func (mvnCmd *mavenCommandStruct) InDirectory(directory string) MavenCommand {
+	mvnCmd.directory = directory
+	return mvnCmd
+}
+
+func (mvnCmd *mavenCommandStruct) SkipTests() MavenCommand {
+	mvnCmd.skipTests = true
+	return mvnCmd
+}
+
+func (mvnCmd *mavenCommandStruct) Profiles(profiles ...string) MavenCommand {
+	mvnCmd.profiles = append(mvnCmd.profiles, profiles...)
+	return mvnCmd
+}
+
+func (mvnCmd *mavenCommandStruct) Execute(targets ...string) (string, error) {
+	var args []string
+
+	// Setup custom maven repository if defined
+	if len(config.GetCustomMavenRepoURL()) > 0 {
+		if err := mvnCmd.setMavenRepoURL(config.GetCustomMavenRepoURL()); err != nil {
+			return "", err
+		}
+	}
+
+	args = append(args, targets...)
+	if len(mvnCmd.profiles) > 0 {
+		args = append(args, fmt.Sprintf("-P%s", strings.Join(mvnCmd.profiles, ",")))
+	}
+	if len(mvnCmd.otherOptions) > 0 {
+		args = append(args, strings.Join(mvnCmd.otherOptions, " "))
+	}
+
+	cmd := CreateCommand(mavenCommandName, args...).InDirectory(mvnCmd.directory)
+
+	// Set logger context if exists
+	if len(mvnCmd.loggerContext) > 0 {
+		cmd.WithLoggerContext(mvnCmd.loggerContext)
+	}
+
+	return cmd.Execute()
+}
+
+func (mvnCmd *mavenCommandStruct) setMavenRepoURL(repoURL string) error {
+	// Create settings.xml in directory
+	if err := ioutil.WriteFile(fmt.Sprintf("%s/settings.xml", mvnCmd.directory), []byte(fmt.Sprintf(settingsFile, repoURL, repoURL)), 0644); err != nil {
+		return err
+	}
+
+	// Add option to get copied settings.xml file
+	mvnCmd.otherOptions = append(mvnCmd.otherOptions, "-ssettings.xml")
```

Comment:
```suggestion
	mvnCmd.otherOptions = append(mvnCmd.otherOptions, "--settings settings.xml")
```

Reply:
I did that in the first place but it does not work as those are 2 args in fact in the end.
And I wanted to avoid to use a map instead of an array as it makes more complicated ...
But we have to use `--settings` absolutely, then I will update the code